addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -788,10 +856,201 @@ private void checkConnection() {
                 markThingOnline();
                 deviceIsConnected = true;
             } else {
-                logger.trace("Connection check failed for thing {} at IP {}", thingID(), commandConnection.getIP());
+                logger.debug("Connection check failed for thing {} at IP {}", thingID(), commandConnection.getIP());
                 disconnect();
             }
         }
+
+        private void startSerialReaders() {
+            if (deviceSupportsSerial1()) {
+                serialReader1 = startSerialReader(CommandType.SERIAL1, ENABLE_TWO_WAY_1, EOM_DELIMITER_1);
+            }
+            if (deviceSupportsSerial2()) {
+                serialReader2 = startSerialReader(CommandType.SERIAL2, ENABLE_TWO_WAY_2, EOM_DELIMITER_2);
+            }
+        }
+
+        private SerialReader startSerialReader(CommandType serialDevice, String enableTwoWayConfig,
+                String eomDelimiterConfig) {
+            Boolean enableTwoWay = (Boolean) thing.getConfiguration().get(enableTwoWayConfig);
+            logger.debug("Enable two-way is {} for thing {} {}", enableTwoWay, thingID(), serialDevice);
+
+            if (enableTwoWay != null && enableTwoWay.equals(Boolean.TRUE)) {
+                // Get the end of message delimiter from the config, URL decode it, and convert it to a byte array
+                String eomString = (String) thing.getConfiguration().get(eomDelimiterConfig);
```

Comment:
Can you replace eom by endOfMessage I know its long, but I had no idea what eom means and do not know what the long version means.